[Linglong] iP

huisthat opened the pull request:
# DukePro 
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://github.com/huisthat/ip).
1. double-click it. 
1. add your tasks.
1. let it manage your tasks for you. :smile: 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public static void main(String[] args) throws IOException {
    new Du("data/du.txt").run();
}
```

Commit:
Add Gradle support

huisthat committed:
level 1

huisthat committed:
clean up level 1

huisthat committed:
level 2

huisthat committed:
level 3

huisthat committed:
clean up level 3

huisthat committed:
complete level 4

huisthat committed:
complete level 4

huisthat committed:
complete level 5 and 6

huisthat committed:
add testing

huisthat committed:
complete Level 7

huisthat committed:
complete level 8

huisthat committed:
resolve branch conflicts

huisthat committed:
improve on OOP

huisthat committed:
add packages

huisthat committed:
add junit tests

huisthat committed:
add javadocs

huisthat committed:
do coding standard level

huisthat committed:
complete level9

huisthat committed:
Merge branch 'branch-A-CodingStandard'

huisthat committed:
Merge branch 'branch-Level-9'

huisthat committed:
clean up javadocs

huisthat committed:
Merge remote-tracking branch 'origin/add-gradle-support'

huisthat committed:
complete level-10

pbthang commented on src/main/java/Du/Parser.java:
```diff
@@ -0,0 +1,117 @@
+    /**
+     * takes in commands from user and parse them
+     */
+    public void parse() {
```
I think the responsibility of Parser should only be parsing 1 command at a time, and the loop should be in Duke class. What do you think?

pbthang commented on src/main/java/Du/Event.java:
```diff
@@ -0,0 +1,52 @@
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd yyyy, HHmm");
+        return "[E][" + this.getStatus() + "] " + this.name
+                + "(at: " + time.getMonth() + " " + time.format(formatter) + ")";
```
In this case, should we use `String.format()` because I find it less complicated than string concatenation?

pbthang commented on data/du.txt:
```diff
@@ -0,0 +1,5 @@
+D , 0 , sth else  , 2021-08-22 18:30
+T , 0 , yay
+T , 0 , hi
+D , 0 ,  , 2021-08-16 16:30
```
Should we put the data file in `.gitignore`, because I think each person needs their own data?

pbthang commented on src/main/java/Du/Task.java:
```diff
@@ -0,0 +1,70 @@
+/**
+ * Class for each individual task and a list of the tasks
+ */
+public class Task {
```
```suggestion
public abstract class Task {
```

pbthang reviewed (commented):
Sorry wrong PR :))

sj1999-BIT commented on src/main/java/Du/TaskList.java:
```diff
@@ -80,4 +80,29 @@ public void find_finished_task(int order) {
+    }
+
+    public void print(ArrayList<Task> arraylist) {
+        System.out.println("____________________________________________________________\n" +
```
This is what I did at first as well. It turns out that a more recommended approach is to set a global final String variable that contains this line. It could potential make the code cleaner.

sj1999-BIT commented on src/main/java/Du/Deadline.java:
```diff
@@ -19,14 +19,14 @@ public Deadline(String name, boolean done, TaskList tasklist, LocalDateTime time
 
     /**
-     * record the Deadline in a certain format to save to the file
+     * records the Deadline in a certain format to save to the file
```
One of the JavaDocs requirements is that every function descriptor should start with a capital letter .It is recommended to follow the general guidelines

sj1999-BIT commented on src/main/java/Du/Du.java:
```diff
@@ -25,7 +28,7 @@ public Du(String filepath) {
 
 
     /**
-     * run Du itself
```
Generally speaking, function descriptors should be more specific and more detailed, unless the function is a very generic one such as getters and setters. I would recommend adding in more details such as what purpose exactly do run do, perhaps 2 to 3 lines will do?

sj1999-BIT commented:
Overall the logic is pretty well done. The presentation could be improved, but then again my own work is also not that presentable XP. Jiayou!

huisthat committed:
rename branch

huisthat committed:
make changes according to reviews

huisthat committed:
add assertions

huisthat committed:
improve code quality

huisthat committed:
Merge pull request #1 from huisthat/A-Assertions

add assertions

huisthat committed:
clean up code

huisthat committed:
make minor changes

huisthat committed:
minor changes

huisthat committed:
minor changes

huisthat committed:
minor changes

huisthat committed:
Merge pull request #4 from huisthat/branch-A-CodeQuality

code quality

huisthat committed:
add recurring task

huisthat committed:
Edit minor details

Code quality needs to be improved.

It allows the code to be viewed better for others.

luluyousef commented on src/main/java/Du/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+
+public class TaskList {
+
+    protected static ArrayList<Task> list_of_tasks = new ArrayList<>();
```
can consider changing the name to listOfTasks or taskList instead to follow the camelCase coding standard! :)

luluyousef commented:
Overall, your code looks pretty good to me. If you want you can consider changing your variables to camelCase format, but good job so far!! 💯

huisthat committed:
Update README.md

huisthat committed:
add better GUI

huisthat committed:
add screenshot

huisthat committed:
Set theme jekyll-theme-modernist

huisthat committed:
add user guide

huisthat committed:
Merge branch 'master' of github.com:huisthat/ip

huisthat committed:
edit user guide

huisthat committed:
fix bugs

huisthat committed:
Fix bugs
